### PR TITLE
[RA2] Traceability between chapter02 and chapter04

### LINKED ISSUE #2120: [RA2 Ch02/Ch04] Ensure full traceability between ch02 and ch04.
Prior to release launch.

### PR DESCRIPTION
**Pull request by tomkivlin**
Resolves #2120

### CODE REVIEW

**Comment by tomkivlin** [comment]
Please don't delete the branch, so I can cherry-pick into stable/elbrus.

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 1]
```diff
@@ -46,28 +46,28 @@ To ensure alignment with the infrastructure profile catalogue, the following req
...
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.002 | Max memory in MB that can be assigned to a single Pod by the Cloud Infrastructure | at least 32 GB<sup>(1)</sup> | at least 32 GB<sup>(1)</sup> | [ra2.ch.012](chapter04.md#42-kubernetes-node)|
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.003 | Max storage in GB that can be assigned to a single Pod by the Cloud Infrastructure | at least 320 GB<sup>(1)</sup> | at least 320 GB<sup>(1)</sup> | [ra2.ch.010](chapter04.md#42-kubernetes-node)|
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.004 | Max number of connection points that can be assigned to a single Pod by the Cloud Infrastructure | 6 | 6 | [ra2.ntw.003](chapter04.md#45-networking-solutions) |
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:52]: Actually there is an inconsistency.
e.cap.004 specifies 6 as the maximum number of connection points while ra2.ntw.003 specifies **at least** 5 additional (to he default) connection points, viz. more than 5 additional connection points can be assigned or a total greater than 6.

Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter02.md:52]: Good point, I will create an issue to rectify.

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 2]
```diff
@@ -46,28 +46,28 @@ To ensure alignment with the infrastructure profile catalogue, the following req
...
-| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.013 | SR-IOV over PCI-PT | Not required | Must support | |
-| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.014 | Hardware coprocessor support (GPU/NPU) | Not required | Not required | |
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.013 | SR-IOV over PCI-PT | Not required | Must support | [ra2.ch.002](chapter04.md#42-kubernetes-node)<br>[ra2.ch.003](chapter04.md#42-kubernetes-node)<br>[ra2.k8s.007](chapter04.md#43-kubernetes)<br>[ra2.ntw.004](chapter04.md#45-networking-solutions)<br>[ra2.ntw.008](chapter04.md#45-networking-solutions)|
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.014 | Hardware coprocessor support (GPU/NPU) | Not required | Not required | [ra2.k8s.007](chapter04.md#43-kubernetes)|
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:62]: GPU support is not required and hence no traceability is required. Also, [ra2.k8s.007](chapter04.md#43-kubernetes) is generic and not for GPU/NPU.

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 3]
```diff
@@ -46,28 +46,28 @@ To ensure alignment with the infrastructure profile catalogue, the following req
...
 | [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.015 | SmartNICs | Not required | Optional | |
-| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.016 | FPGA/other Acceleration H/W | Not required | Optional | |
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.016 | FPGA/other Acceleration H/W | Not required | Optional | [ra2.k8s.006](chapter04.md#43-kubernetes)<br>[ra2.ntw.012](chapter04.md#45-networking-solutions)|
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:64]: should it be ra2.k8s.007 (Device PlugIns) and not ra2.k8s.006 (NUMA support)
```suggestion
| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | e.cap.016 | FPGA/other Acceleration H/W | Not required | Optional | [ra2.k8s.007](chapter04.md#43-kubernetes)<br>[ra2.ntw.012](chapter04.md#45-networking-solutions)|
```

File: doc/ref_arch/kubernetes/chapters/chapter02.md (hunk 1 above)
Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter02.md:52]: #2207 created

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 4]
```diff
@@ -46,28 +46,28 @@ To ensure alignment with the infrastructure profile catalogue, the following req
...
 | [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | *e.cap.017* | *Ability to monitor L2-L7 data from workload* | *n/a<sup>(3)</sup>* | *n/a<sup>(3)</sup>* | |
-| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | i.cap.014 | Specifies the proportion of CPU cores consumed by the Cloud Infrastructure system on the worker nodes. If SMT is used, it indicates the number of consumed SMT threads. | 2 | 2 | |
+| [4.2.5](../../../ref_model/chapters/chapter04.md#425-cloud-infrastructure-profile-capabilities-mapping) | i.cap.014 | Specifies the proportion of CPU cores consumed by the Cloud Infrastructure system on the worker nodes. If SMT is used, it indicates the number of consumed SMT threads. | 2 | 2 | [ra2.k8s.008](chapter04.md#43-kubernetes) |
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:66]: Nit
Not sure how to handle this. i.cap.014 "proportion of CPU cores consumed ..." while ra2.k8s.008 "... avoid resource starvation issues on nodes, reserve compute resources ...". Maybe a note that states the #reserved cores indirectly provides the proportion.

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 5]
```diff
@@ -133,18 +133,18 @@ The required number of connection points to a Pod is described in `e.cap.004` ab
...
-| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.pci.cfg.002 | PCIe speed | Gen 3 | Gen 3 | |
-| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.pci.cfg.003 | PCIe Lanes | 8 | 8 | |
+| [5.4.2](../../../ref_model/chapters/chapter05.md#542-storage-configurations) | infra.hw.stg.ssd.cfg.002 | Local Storage SSD | Should support | Should support |[ra2.ch.009](chapter04.md#42-kubernetes-node)|
+| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.nic.cfg.001 | Total Number of NIC Ports available in the host | 4 | 4 |[ra2.ch.013](chapter04.md#42-kubernetes-node)|
+| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.nic.cfg.002 | Port speed specified in Gbps (minimum values) | 10 | 25 |[ra2.ch.014](chapter04.md#42-kubernetes-node)<br>[ra2.ch.015](chapter04.md#42-kubernetes-node)|
+| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.pci.cfg.001 | Number of PCIe slots available in the host | 8 | 8 | [ra2.ch.016](chapter04.md#42-kubernetes-node) |
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:145]: typo
```suggestion
| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.pci.cfg.001 | Number of PCIe slots available in the host | 8 | 8 | [ra2.ch.015](chapter04.md#42-kubernetes-node) |
```

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 6]
```diff
@@ -133,18 +133,18 @@ The required number of connection points to a Pod is described in `e.cap.004` ab
...
+| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.pci.cfg.002 | PCIe speed | Gen 3 | Gen 3 |[ra2.ch.016](chapter04.md#42-kubernetes-node) |
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:146]:
```suggestion
| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.pci.cfg.002 | PCIe speed | Gen 3 | Gen 3 |[ra2.ch.015](chapter04.md#42-kubernetes-node) |
```

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 7]
```diff
@@ -133,18 +133,18 @@ The required number of connection points to a Pod is described in `e.cap.004` ab
...
+| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.pci.cfg.003 | PCIe Lanes | 8 | 8 |[ra2.ch.016](chapter04.md#42-kubernetes-node) |
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:147]:
```suggestion
| [5.4.3](../../../ref_model/chapters/chapter05.md#543-network-resources) | infra.hw.pci.cfg.003 | PCIe Lanes | 8 | 8 |[ra2.ch.015](chapter04.md#42-kubernetes-node) |
```

File: doc/ref_arch/kubernetes/chapters/chapter02.md [hunk 8]
```diff
@@ -296,35 +296,35 @@ The requirements in this section are to be delivered in addition to those in [se
...
 | `req.kcm.gen.01` | Kubernetes Cluster | General | The Architecture **must** support policy driven horizontal auto-scaling of Kubernetes cluster. | |
-| `req.kcm.gen.02` | Kubernetes Cluster | General | The Architecture **must** enable workload resiliency. | |
-| `req.int.api.01` | API | General | The Architecture **must** leverage the Kubernetes APIs to discover and declaratively manage compute (virtual and bare metal resources), network, and storage. | |
-| `req.int.api.02` | API | General | The Architecture **must** support the usage of a Kubernetes Application package manager using the Kubernetes API, like Helm v3. | |
+| `req.kcm.gen.02` | Kubernetes Cluster | General | The Architecture **must** enable workload resiliency. |[ra2.k8s.004](chapter04.md#43-kubernetes)|
+| `req.int.api.01` | API | General | The Architecture **must** leverage the Kubernetes APIs to discover and declaratively manage compute (virtual and bare metal resources), network, and storage. |[ra2.ntw.001](chapter04.md#45-networking-solutions)<br>[ra2.ntw.008](chapter04.md#45-networking-solutions)<br>[ra2.app.006](chapter04.md#49-kubernetes-workloads)|
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter02.md:326]: Why ra2.ntw.008 - SR-IOV?
also we need for both compoute and storage -- CRI, CSI?

Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter02.md:326]: ra2.ntw.008 is about Device Plugin, which is about leveraging k8s APIs (DPAPI being one). Less about the SR-IOV part.

File: doc/ref_arch/kubernetes/chapters/chapter02.md (hunk 4 above)
Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter02.md:66]: It's only proportional if SMT isn't enabled - I've added a clarification in https://github.com/cntt-n/CNTT/pull/2204/commits/eafe3febe3fff7a65c51afa0265217d6365bfea0, does that work?

File: doc/ref_arch/kubernetes/chapters/chapter02.md (hunk 2 above)
Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter02.md]: Good point - have corrected in https://github.com/cntt-n/CNTT/pull/2204/commits/5afc2c99032e74ccbd2e82ba9a8dbf53d42e5f2c

File: doc/ref_arch/kubernetes/chapters/chapter02.md (hunk 8 above)
Comment (pgoyal01) [reply doc/ref_arch/kubernetes/chapters/chapter02.md:326]: @tomkivlin Sorry wasn't clear. 
| `req.int.api.01` | API | General | The Architecture **must** leverage the Kubernetes APIs to discover and declaratively manage compute (virtual and bare metal resources), network, and storage.
-- | -- | --|-- 

The compute and storage are not addressed; only network is.


ra2.ntw.008 | SR-IOV Device Plugin for Network Intensive | When hosting workloads that match the Network Intensive profile and  require SR-IOV acceleration, a Device Plugin for SR-IOV must be used to  configure the SR-IOV devices and advertise them to the kubelet.
-- | -- | --

Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter02.md]: Oh I see, yes agreed. I've added a clarification in https://github.com/cntt-n/CNTT/pull/2204/commits/9c7c228c3a70e396d17293a46f470804172e75b0.

**Review by pgoyal01 — APPROVED** [review]

**Review by rgstori — APPROVED** [review]

**Comment by tomkivlin** [comment]
@CsatariGergely @peterwoerndle @petorre @ulikleber could one of you give this a once-over please? Just need one more approval for merging.  Thanks!

**Review by collivier — APPROVED** [review]

**Pull request merged**